graylog2-web-interface/src/components/streamrules/StreamRule.jsx

```diff
@@ -1,96 +1,109 @@
-          this.props.onDelete(this.props.streamRule.id);
+      StreamRulesStore.remove(stream.id, streamRule.id, () => {
+        if (onDelete) {
+          onDelete(streamRule.id);
         }
         UserNotification.success('Stream rule has been successfully deleted.', 'Success');
       });
     }
-  },
+  };
 
-  _onSubmit(streamRuleId, data) {
-    StreamRulesStore.update(this.props.stream.id, streamRuleId, data, () => {
-      if (this.props.onSubmit) {
-        this.props.onSubmit(streamRuleId, data);
+  const _onSubmit = (streamRuleId, data) => {
+    StreamRulesStore.update(stream.id, streamRuleId, data, () => {
+      if (onSubmit) {
+        onSubmit(streamRuleId, data);
       }
       UserNotification.success('Stream rule has been successfully updated.', 'Success');
     });
-  },
+  };
 
-  _formatActionItems() {
+  const _formatActionItems = () => {
     return (
-      <span>
-        <a href="#" onClick={this._onDelete} style={{ marginRight: 5 }}>
+      <ActionButtonsWrap>
+        <Button bsStyle="link"
+                bsSize="xsmall"
+                onClick={_onDelete}>
           <Icon name="trash-o" />
-        </a>
-        <a href="#" onClick={this._onEdit} style={{ marginRight: 5 }}>
+        </Button>
+        <Button bsStyle="link"
+                bsSize="xsmall"
+                onClick={_onEdit}>
           <Icon name="edit" />
-        </a>
-      </span>
-    );
-  },
-
-  _getMatchDataClassNames() {
-    return (this.props.matchData.rules[this.props.streamRule.id] ? 'alert-success' : 'alert-danger');
-  },
-
-  render() {
-    const { streamRule } = this.props;
-    const { streamRuleTypes } = this.props;
-    const actionItems = (this.isPermitted(this.props.permissions, [`streams:edit:${this.props.stream.id}`]) ? this._formatActionItems() : null);
-    const className = (this.props.matchData ? this._getMatchDataClassNames() : null);
-    const description = this.props.streamRule.description ? <small>{' '}({this.props.streamRule.description})</small> : null;
-    return (
-      <li className={className}>
-        {actionItems}
-        <HumanReadableStreamRule streamRule={streamRule} streamRuleTypes={streamRuleTypes} />
-        <StreamRuleForm ref={(streamRuleForm) => { this.streamRuleForm = streamRuleForm; }}
-                        streamRule={streamRule}
-                        streamRuleTypes={streamRuleTypes}
-                        title="Edit Stream Rule"
-                        onSubmit={this._onSubmit} />
-        {description}
-      </li>
+        </Button>
+      </ActionButtonsWrap>
     );
-  },
-});
+  };
+
+  const matchDataStyle = () => (matchData.rules[streamRule.id] ? 'success' : 'danger');
+  const actionItems = isPermitted(permissions, [`streams:edit:${stream.id}`]) ? _formatActionItems() : null;
+  const description = streamRule.description ? <small>{' '}({streamRule.description})</small> : null;
+
+  return (
+    <ListGroupItem bsStyle={!isEmpty(matchData) && matchDataStyle()}>
```

Comment:
This throws a warning because the value of `bsStyle` can be a boolean and it is not the right prop type. We should replace `&&` with a conditional operator.